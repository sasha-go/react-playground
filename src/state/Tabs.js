import React from "react";

class Tabs extends React.Component {
	static defaultProps = {
		tabs: []
	};

	state = {
		currentTabIndex: 0
	};

	handleButtonClick(index) {
    this.setState({ currentTabIndex: index })
	}
	
  renderButtons() {
		return this.props.tabs.map((tab, index) => (
      <button key={index} onClick={() => this.handleButtonClick(index)}>
        {tab.name}
      </button>
    ))
  }

	renderContent() {
		const currentTab = this.props.tabs[this.state.currentTabIndex]
		return (
			<div className='content'>
				{currentTab.content}
			</div>
		)	
	}

	render() {
		return (
			<div>
				{this.renderButtons()}
				{!!this.props.tabs.length && this.renderContent()}
			</div>
		)
	}
}

export default Tabs;
