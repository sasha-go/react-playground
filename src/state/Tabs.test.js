import React from 'react';
import ReactDOM from 'react-dom';
import Tabs from './Tabs';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe(`Tabs Component`, () => {
	const tabsProp = []

  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Tabs />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

	it('renders empty given no tabs', () => {
		const wrapper = shallow(<Tabs />)
		expect(toJson(wrapper)).toMatchSnapshot();
	})

	it('renders the first tab by default', () => {
		const wrapper = shallow(<Tabs tabs={tabsProp}/>)
		expect(toJson(wrapper)).toMatchSnapshot()
	})

	it('closes the first tab and opens any clicked tab', () => {
		const wrapper = shallow(<Tabs tabs={tabsProp} />)
		wrapper.find('button').at(1).simulate('click')
		expect(toJson(wrapper)).toMatchSnapshot()
	})

	// it('closes the first tab and opens any clicked tab', () => {
	// 	const wrapper = shallow(<Tabs tabs={tabsProp} />)
	// 	wrapper.find('button').at(1).simulate('click')
	// 	expect(toJson(wrapper)).toMatchSnapshot()
	// })
})

