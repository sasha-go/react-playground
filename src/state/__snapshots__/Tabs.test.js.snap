// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Component closes the first tab and opens any clicked tab 1`] = `<div />`;

exports[`Tabs Component renders empty given no tabs 1`] = `<div />`;

exports[`Tabs Component renders the first tab by default 1`] = `<div />`;
